pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -637,25 +673,18 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         uint256[] memory endWeights,
         IERC20[] memory tokens
     ) internal virtual {
+        uint256 totalWeight = _totalWeight;
         uint256 normalizedSum = 0;
-        bytes32 tokenState;
 
         for (uint256 i = 0; i < endWeights.length; i++) {
             uint256 endWeight = endWeights[i];
             _require(endWeight >= WeightedMath._MIN_WEIGHT, Errors.MIN_WEIGHT);
+            normalizedSum = normalizedSum.add(endWeight);
```

Comment:
If these are normalized, that should be made clear. I'd call these `normalizedStartWeights`, `normalizedWeight`, etc.

Reply:
Also it seems a bit odd and inefficient to normalize everything and then denormalize. Why not just check that the sum adds up to `_totalWeight` instead? That also means we won't have weird precision issues in the normalization/denormalization process.

Reply:
Usually when this gets called, the values are passed in from outside: in the constructor, or a call to updateWeightsGradually, and those we do want to sum to 1, to hide the internal normalization.

The only place it normalizes and denormalizes again is when you call updateWeightsGradually.

I can't avoid normalizing, because normalized weights are needed for the event.

Neither can I avoid denormalizing by simply copying compressed and denormalized end weights to start weights, since it could be in the middle of an ongoing change.

This kind of thing is why I thought it would be best to extend end weights to 64 bits so they both have the same precision.